server/runner/backuprun/runner.go

```diff
@@ -315,14 +316,30 @@ func (r *Runner) startAutoBackups(ctx context.Context) {
 		if _, ok := r.stateCfg.RunningBackupDatabases.Load(backupSetting.DatabaseID); ok {
 			continue
 		}
-		db := backupSetting.Database
-		if db.Name == api.AllDatabaseName {
+		database, err := r.store.GetDatabaseV2(ctx, &store.FindDatabaseMessage{UID: &backupSetting.DatabaseID})
+		if err != nil {
+			log.Error("Failed to get database", zap.Error(err))
+			return
+		}
+		if database.DatabaseName == api.AllDatabaseName {
 			// Skip backup job for wildcard database `*`.
 			continue
 		}
-		backupName := fmt.Sprintf("%s-%s-%s-autobackup", api.ProjectShortSlug(db.Project), api.EnvSlug(db.Instance.Environment), t.Format("20060102T030405"))
+		project, err := r.store.GetProjectV2(ctx, &store.FindProjectMessage{ResourceID: &database.ProjectID})
+		if err != nil {
+			log.Error("Failed to get project", zap.Error(err))
+		}
+		instance, err := r.store.GetInstanceV2(ctx, &store.FindInstanceMessage{EnvironmentID: &database.EnvironmentID, ResourceID: &database.InstanceID})
+		if err != nil {
+			log.Error("Failed to get instance", zap.Error(err))
+		}
+		environment, err := r.store.GetEnvironmentV2(ctx, &store.FindEnvironmentMessage{ResourceID: &instance.EnvironmentID})
```

Comment:
Do we need to check if the project and environment is deleted here?

Reply:
Done.